linker/shared/src/main/scala/org/scalajs/linker/frontend/BaseLinker.scala

```diff
@@ -228,9 +230,12 @@ private[frontend] object BaseLinker {
 
     field match {
       case field: FieldDef =>
-        if (field.flags.namespace.isStatic) classInfo.isAnyStaticFieldUsed
-        else if (classInfo.kind.isJSType) classInfo.isAnyPrivateJSFieldUsed
-        else classInfo.isAnySubclassInstantiated
+        if (field.flags.namespace.isStatic)
+          classInfo.staticFieldsRead(field.name.name) || classInfo.staticFieldsWritten(field.name.name)
+        else if (classInfo.kind.isJSClass || classInfo.isAnySubclassInstantiated)
```

Comment:
This mirrors the previous behavior that (non static) JS fields are kept even if there is no instance. However, I do not understand why this is necessary. @sjrd ?